### PR TITLE
Guest Address Space Changed: 1GB->4GB & SPSR_irq saved in guest contexts

### PR DESCRIPTION
**Pull request by simonkim**
- Stage 2 IPA:PA mapping for Guest Address Space Changed
- context.c: saves spsr_irq, sp_irq, lr_irq in a guest context to safely preempt guest running in IRQ mode

Guest Address Space: 1GB -> 4GB
- 0x0000_0000 ~ 1GB Peripherals
- 0x4000_0000 ~ 1GB Unmapped
- 0x8000_0000 ~ 1GB RAM, code entry, 256MB ~ 1GB mapped memory
- 0xC000_0000 ~ 1GB Unmapped
- Guest 0: 768MB RAM
- Guest 1: 256MB RAM
- Guest Code Entry Changed: 0x0000_0000 -> 0x8000_0000

### CODE REVIEW

File: monitor/context.c [hunk 1]
```diff
@@ -145,28 +260,25 @@ hvmm_status_t context_perform_switch(void)
...
-
     if ( _current_guest_vmid == VMID_INVALID ) {
+        printh("context: launching the first guest\n");
         /* very first time, to the default first guest */
         result = context_perform_switch_to_guest_regs( 0, _next_guest_vmid );
         /* DOES NOT COME BACK HERE */
     } else if ( _next_guest_vmid != VMID_INVALID && _current_guest_vmid != _next_guest_vmid ) {
         struct arch_regs *regs = trap_saved_regs();
         if ( (regs->cpsr & 0x1F) != 0x1A ) {
+            uart_print( "curr:" ); uart_print_hex32( _current_guest_vmid ); uart_print( "\n\r" );
```
Comment (Wonseok) [inline monitor/context.c:271]: Why is this code still using uart_print? :D

File: monitor/context.c [hunk 2]
```diff
@@ -145,28 +260,25 @@ hvmm_status_t context_perform_switch(void)
...
+            uart_print( "next:" ); uart_print_hex32( _next_guest_vmid ); uart_print( "\n\r" );
```
Comment (Wonseok) [inline monitor/context.c:272]: Same as the above

**Comment by Wonseok** [comment]
It looks very good :D

File: monitor/context.c (hunk 1 above)
Comment (simonkim) [inline monitor/context.c:271]: because changing that line was not the point of this commit.

File: monitor/context.c (hunk 2 above)
Comment (simonkim) [inline monitor/context.c:272]: the same pointless comment.

**Pull request merged**